.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -11,107 +11,197 @@
+# node:22.1-alpine
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/node@sha256:487dc5d5122d578e13f2231aa4ac0f63068becd921099c4c677c850df93bede8 as owlbot-cli-build
+ARG OWLBOT_CLI_COMMITTISH=ac84fa5c423a0069bbce3d2d869c9730c8fdf550
+
+# install tools
+RUN apk add git
+
+# Clone the owlbot-cli source code
+WORKDIR /tools
+RUN git clone https://github.com/googleapis/repo-automation-bots
+WORKDIR /tools/repo-automation-bots/packages/owl-bot
+RUN git checkout "${OWLBOT_CLI_COMMITTISH}"
+
+# Part of the code path (that we don't use) ends up touching a dependency called
+# @google-cloud/datastore that tries a fs.readFileSync that is not handled by
+# default by esbundle (esbundle is good a figuring out imports but doesn't
+# actively scan filesystem interactions such as fs.readFileSync). This makes the
+# app to fetch a file at runtime that is not available in the bundle context.
+# This is why we remove this import and its usage from the entrypoint.
+RUN sed -i '/testWebhook/d' src/bin/owl-bot.ts
+
+# Bundle the source code and its dependencies into a single javascript file
+# with all its dependencies embedded.
+# This is because SEA (see below) cannot
+# resolve external modules in a multi-file project.
+# We use the esbuild tool. See https://esbuild.github.io/
+COPY ./.cloudbuild/library_generation/image-configuration/owlbot-cli-build-config.js .
+RUN npm i esbuild
+RUN node owlbot-cli-build-config.js
+
+# Compile the bundled javascript file into a Linux executable
+# Create a Standalone Executable Application (SEA) configuration file.
+# See https://nodejs.org/api/single-executable-applications.html
+RUN echo '{ "main": "bundle.js", "output": "sea-prep.blob" }' > build/sea-config.json
+WORKDIR /tools/repo-automation-bots/packages/owl-bot/build
+RUN node --experimental-sea-config sea-config.json
+RUN cp $(command -v node) owl-bot-bin
+RUN npx postject owl-bot-bin NODE_SEA_BLOB sea-prep.blob \
+    --sentinel-fuse NODE_SEA_FUSE_fce680ab2cc467b6e072b8b5df1996b2
+
+# move to a simple path for convenience
+RUN cp ./owl-bot-bin /owl-bot-bin
+
+# Creates the generator jar
+# maven:3.8.6-openjdk-11-slim
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/maven@sha256:2cb7c73ba2fd0f7ae64cfabd99180030ec85841a1197b4ae821d21836cb0aa3b  as ggj-build
 
 WORKDIR /sdk-platform-java
 COPY . .
 # {x-version-update-start:gapic-generator-java:current}
 ENV DOCKER_GAPIC_GENERATOR_VERSION="2.45.1-SNAPSHOT" 
 # {x-version-update-end}
 
-RUN mvn install -B -ntp -DskipTests -Dclirr.skip -Dcheckstyle.skip
-RUN cp "/root/.m2/repository/com/google/api/gapic-generator-java/${DOCKER_GAPIC_GENERATOR_VERSION}/gapic-generator-java-${DOCKER_GAPIC_GENERATOR_VERSION}.jar" \
-  "./gapic-generator-java.jar"
+# use Docker Buildkit caching for faster local builds
+RUN --mount=type=cache,target=/root/.m2 mvn install -B -ntp -T 1.5C \
+    -DskipTests -Dclirr.skip -Dcheckstyle.skip -Djacoco.skip -Dmaven.test.skip \
```

Comment:
Why we need both `-DskipTests` and `-Dmaven.test.skip`?

Reply:
Actually `skipTests` is redundant since it only skips the tests execution, whereas `test.skip` skips both the compilation and execution. I removed it. Thanks for the catch.